extensions/common/context.cc

```diff
@@ -551,5 +554,21 @@ bool sanitizeBytes(std::string* buf) {
   return modified;
 }
 
+// Used for `destination_service` fallback. Unlike elsewhere when that fallback
+// to workload name, this falls back to "unknown" when the canonical name label
+// is not found. This preserves the existing behavior for `destination_service`
+// labeling. Using a workload name as a service name could be potentially
+// problematic.
+std::string getServiceNameFallback() {
+  auto buf = getProperty({"node", "metadata", "LABELS"});
+  if (buf.has_value()) {
+    for (const auto& [key, val] : buf.value()->pairs())
+      if (key == ::Wasm::Common::kCanonicalServiceLabelName.data()) {
```

Comment:
Why is canonical service name the right thing to use? For the majority of workloads there is 1 Service per workload, which we encode in the cluster metadata. Would it be better to use that?

Reply:
This happen on inbound passthrough, destination service cannot be decided.